consensus.go

```diff
@@ -378,23 +394,22 @@ func (p *Pbft) runValidateState(ctx context.Context) { // start new round
 			span.AddEvent("Commit")
 		}
 	}
+  
+	timeout := p.roundTimeout(p.state.view.Round)
 
-	timeout := p.exponentialTimeout()
 	for p.getState() == ValidateState {
 		_, span := p.tracer.Start(ctx, "ValidateState")
 
 		msg, ok := p.getNextMessage(span, timeout)
-		if !ok {
+		if !ok || msg == nil {
```

Comment:
Please revert this change. I usually have a hard time myself with weird logic operands so I avoid them as much as possible.